packages/core/src/api/config/graphql-custom-fields.ts

```diff
@@ -185,7 +185,10 @@ export function addGraphQLCustomFields(
         }
     }
 
-    if (customFieldConfig.Address?.length) {
+    const publicAddressFields = customFieldConfig.Address?.filter(
+        config => !config.internal && (publicOnly === true ? config.public !== false : true),
+    );
+    if (customFieldConfig.Address?.length && publicAddressFields?.length) {
```

Comment:
I think the `customFieldConfig.Address?.length` check can be removed now right? The `publicAddressFields` value would already cover this.

Reply:
Actually I'll just refactor this separately, no worries.